### PR TITLE
fix: preset-vue

### PR DESCRIPTION
**Pull request by jeffwcx**
### 🤔 这个变动的性质是？/ What is the nature of this change?

- [ ] 新特性提交 / New feature
- [x] bug 修复 / Fix bug
- [ ] 样式优化 / Style optimization
- [ ] 代码风格优化 / Code style optimization
- [ ] 性能优化 / Performance optimization
- [ ] 构建优化 / Build optimization
- [ ] 网站、文档、Demo 改进 / Website, documentation, demo improvements
- [ ] 重构代码或样式 / Refactor code or style
- [ ] 测试相关 / Test related
- [ ] 其他 / Other

### 🔗 相关 Issue / Related Issue
#2141

### 💡 需求背景和解决方案 / Background or solution

+ preset-vue引入的问题
   + webpack 配置导致的`@makotot/ghostui/src/Scrollspy` 引入问题（即从node_modules引入tsx）
   更改为
   ```ts
   import { Scrollspy as ScrollSpy  } from "@makotot/ghostui"
   ```
 
   + repository 空字段问题
+ mfsu 依赖搜集机制所导致的vue及外部依赖产生多个实例

```js
mfsu: {
 shared: {
   ...msfuConfig.shared,
   vue: { singleton: true },
 },
},
```
+ 每个demo传给live compiler的filename需唯一
+ 最新版的vue的泛型组件类型有更改，需要支持

@PeachScript @Jinbao1001 

### 📝 更新日志 / Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix preset-vue  |
| 🇨🇳 Chinese |   修复preset-vue   |

### CODE REVIEW

**Comment by Jinbao1001** [comment]
+1

**Review by PeachScript — CHANGES_REQUESTED** [review]

File: src/loaders/markdown/index.ts [hunk 1]
```diff
@@ -188,7 +188,7 @@ export const demos = {
             // omit entry file
             ...(key !== entryFileName
               ? {
-                  [key]: `{{{require('${path}')}}}`,
+                  [key]: `{{{import('${path}')}}}`,
```
[the remark this answers is not in the document]
Comment (jeffwcx) [reply src/loaders/markdown/index.ts:191]: 这里暂时的异步没关系吧，`expandDemoContext` 那里会await一下。不必要的chunk拆分是什么？

File: src/loaders/markdown/index.ts [hunk 2]
```diff
@@ -68,6 +69,12 @@ function getDemoSourceFiles(demos: IMdTransformerResult['meta']['demos'] = []) {
   }, []);
 }
 
+const resolver = enhancedResolve.create.sync({
+  mainFields: ['browser', 'module', 'main'],
+  extensions: ['.js', '.jsx', '.ts', '.tsx', '.json'],
+  conditionNames: ['import', 'default', 'browser'],
```
Comment (Jinbao1001) [inline src/loaders/markdown/index.ts:75]: conditionNames要不要和项目已有保持一致: ['import', 'require', 'default', 'browser', 'node']

**Comment by lifegit** [comment]
+1, currently still exists.

fatal - TypeError: Cannot read properties of undefined (reading 'directory')
    at Hook.fn (/ui-pro/node_modules/.pnpm/@dumijs+preset-vue@2.3.0_dumi@2.4.7_@babel+core@7.24.7_@swc+helpers@0.5.1_@types+node@20.5.1__s4uoca4yeixx4ez3l74iass7zu/node_modules/@dumijs/preset-vue/dist/common.js:42:91)
    at /ui-pro/node_modules/.pnpm/@umijs+core@4.3.10/node_modules/@umijs/core/dist/service/service.js:136:38
    at _next10 (eval at create (/ui-pro/node_modules/.pnpm/@umijs+bundler-utils@4.3.10/node_modules/@umijs/bundler-utils/compiled/tapable/index.js:1:8410), <anonymous>:85:18)
    at eval (eval at create (/ui-pro/node_modules/.pnpm/@umijs+bundler-utils@4.3.10/node_modules/@umijs/bundler-utils/compiled/tapable/index.js:1:8410), <anonymous>:110:1)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
fatal - A complete log of this run can be found in:
fatal - /ui-pro/myapp/node_modules/.cache/logger/umi.log
fatal - Consider reporting a GitHub issue on https://github.com/umijs/umi/issues

**Pull request merged**